### DISCUSSION TITLE
Idea: RFC 5805 Support

### DISCUSSION THREAD

**Discussion by JesseCoretta** [OpenIdentityPlatform/OpenDJ#428] [Ideas]
Hello.

I would like to hear some feedback on the topic of [RFC 5805](https://datatracker.ietf.org/doc/html/rfc5805) support in OpenDJ. At present, there is no mention of this standard in our source code, nor do I see any issues (open or closed) which discuss or make reference to this standard.

For those unaware, I'll cite [Section 1 of RFC 5805](https://datatracker.ietf.org/doc/html/rfc5805#section-1) for descriptive reasons:

> This document extends the Lightweight Directory Access Protocol (LDAP) [RFC4510](https://datatracker.ietf.org/doc/html/rfc4510) to allow clients to relate a number of update operations [RFC4511](https://datatracker.ietf.org/doc/html/rfc4511) and have them performed as one unit of    interaction, a transaction.  As with distinct update operations, each transaction has atomic, consistency, isolation, and durability ([ACID](https://datatracker.ietf.org/doc/html/rfc5805#ref-ACID)) properties.
> 
> This extension consists of two extended operations, one control, and one unsolicited notification message.  The Start Transaction operation is used to obtain a transaction identifier.  This identifier is then attached to multiple update operations to indicate that they belong to the transaction using the Transaction Specification control.  The End Transaction is used to settle (commit or abort) the transaction.  The Aborted Transaction Notice is provided by the server to notify the client that the server is no longer willing or able to process an outstanding transaction.

Is this something you would like to see researched for potential implementation in the future?  Speaking for myself only, I would eagerly vote in favor of this. Everyone may participate in this discussion, maintainers and users alike, **so please do chime in!**

As I understand it, implementing this functionality would take a fair amount of research (which I am happy to do) and would involve non-trivial changes to the server source and (possibly) the client.  Fortunately there are a few directory products on the market that have implemented this feature, so at least we have some reference material to leverage beyond the RFC alone.

I await your thoughts. Thank you.

Jesse 😃

**Comment by vharseko** [comment]
please make [Feature request](https://github.com/OpenIdentityPlatform/OpenDJ/issues/new?assignees=&labels=&projects=&template=Feature_request.md)  and propose a positive and negative testing scenario

**Comment by JesseCoretta** [comment]
I cannot.  Please see [my final response](https://github.com/OpenIdentityPlatform/OpenDJ/issues/439#issuecomment-2436512146).  Farewell.

**Comment by vharseko** [comment]
https://github.com/OpenIdentityPlatform/OpenDJ/pull/469

**Comment by vharseko** [comment]
[4.10.0+](https://github.com/OpenIdentityPlatform/OpenDJ/releases/tag/4.10.0)
